### PR TITLE
Refactoring code with logging and exception handling

### PR DESCRIPTION
**Pull request by jpamarthi**
1. Bumped the version
2. Refactored the functions get_ssacf, detect_outliers_today_classic, and detect_outliers_latest_classic
3. Added new test cases for the same

### CODE REVIEW

**Comment by aseemanand** [comment]
Hi Jagadish, Code run was failing for the following check for detect_outliers_latest_classic and detect_outliers_today_classic as it’s before date index creation so moved it after the set index -

```
# Validate DataFrame structure
if not isinstance(df_outliers.index, pd.DatetimeIndex):
logger.error("DataFrame index must be DatetimeIndex")
raise DataValidationError("DataFrame must have a DatetimeIndex”)
```

Also, do we need the above check if it’s already set index in _check_and_convert_date_ function?

File: src/pycatcher/catch.py
Comment (aseemanand) [inline src/pycatcher/catch.py:1]: Hi Jagadish, Notebook run was failing for the following check in detect_outliers_latest_classic and detect_outliers_today_classic as it’s before date index creation so moved it after the set index -

# Validate DataFrame structure
if not isinstance(df_outliers.index, pd.DatetimeIndex):
logger.error("DataFrame index must be DatetimeIndex")
raise DataValidationError("DataFrame must have a DatetimeIndex”)

Also, do we need the above check if there's already set index in check_and_convert_date function?

Comment (jpamarthi) [reply src/pycatcher/catch.py:1]: @aseemanand I agree with you. I think it is a redundant check at this point.

**Review by aseemanand — APPROVED** [review]
LGTM

**Review by aseemanand — APPROVED** [review]
LGTM

**Pull request merged**